pkg/testmachinery/testflow/node/node.go

```diff
@@ -169,6 +147,54 @@ func (n *Node) Status() *tmv1beta1.StepStatus {
 	return status
 }
 
+func (n *Node) isRootNode() bool {
+	return n.inputSource != nil
+}
+
+func (n *Node) DetermineArtifacts() []argov1.Artifact {
+	artifactsMap := make(map[string]argov1.Artifact)
+
+	if n.step.UseGlobalArtifacts {
```

Comment:
if we use maps then we could move the global artifact down and remove the if in line 168.
Then we can get rid of one if condition and have a more semantic code that global artifacts are not the default.
```go
if n.isRootNode() {
  artifactMap["kubeconfig"]...
  ...
}
if n.step.UseGlobalArtifacts {
  ...
}
```

Reply:
done